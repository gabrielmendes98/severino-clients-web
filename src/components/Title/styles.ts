import styled from '@emotion/styled';

export const Container = styled.div`
  border-left: 4px solid ${({ theme }) => theme.palette.secondary.main};
  padding-left: ${({ theme }) => theme.spacing(4)};
`;
