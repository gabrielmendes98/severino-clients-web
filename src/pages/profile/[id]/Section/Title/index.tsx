import Text from 'components/Text';

const SectionTitle = ({ title }: { title: string }) => (
  <Text fontWeight="bold" color="secondary" variant="h6">
    {title}
  </Text>
);

export default SectionTitle;
