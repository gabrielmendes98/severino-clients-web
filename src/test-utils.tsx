import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
