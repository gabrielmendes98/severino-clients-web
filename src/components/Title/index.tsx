import Text from 'components/Text';
import { Container } from './styles';

interface Props {
  title: string;
}

const Title = ({ title }: Props) => (
  <Container>
    <Text color="secondary" fontWeight="bold" variant="h5">
      {title}
    </Text>
  </Container>
);

export default Title;
