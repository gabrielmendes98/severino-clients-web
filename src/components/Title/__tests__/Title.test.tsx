import { render } from 'test-utils';
import Title from '../index';

it('should match snapshot', () => {
  const { container } = render(<Title title="test title" />);
  expect(container).toMatchSnapshot();
});
