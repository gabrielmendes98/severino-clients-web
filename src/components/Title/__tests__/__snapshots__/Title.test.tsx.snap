// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="css-17un093"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 css-1wlit04-MuiTypography-root"
    >
      test title
    </h5>
  </div>
</div>
`;
